plc4j/drivers/firmata/src/main/java/org/apache/plc4x/java/firmata/readwrite/protocol/FirmataProtocolLogic.java

```diff
@@ -71,12 +71,22 @@ public void onConnect(ConversationContext<FirmataMessage> context) {
         LOGGER.debug("Sending Firmata Reset Command");
         FirmataMessageCommand resetCommandMessage = new FirmataMessageCommand(new FirmataCommandSystemReset());
         context.sendRequest(resetCommandMessage)
+            .onTimeout(e -> {
+                LOGGER.info("Timeout during Connection establishment, closing channel...");
+            })
+            .onError((firmataMessage, throwable) -> {
+                LOGGER.error("Error during Connection establishment. Got: {}", firmataMessage.toString(), throwable);
+            })
+            // Technically the remote will send two messages, one containing only the version information,
+            // the second one also containing the name of the remote station. We simply wait for the
+            // second one and silently have the decode method drop the first.
             .expectResponse(FirmataMessage.class, REQUEST_TIMEOUT)
-            .only(FirmataMessageCommand.class)
```

Comment:
was there a problem with the `only`?

Reply:
Admittedly I didn't know what "only" was all about and the driver didn't do anything ... so I just changed things to how I used to do them.

Reply:
![image](https://github.com/apache/plc4x/assets/1769155/09d5beeb-b843-4dc7-af4b-956bb1dd0106)